### DISCUSSION TITLE
New to tract-onnx

### DISCUSSION THREAD

**Discussion by theyash14** [sonos/tract#1534] [Q&A]
I'm coming from ort library in rust [Due to no support for wasm].
Before I code for wasm, I wanted to run the code normally.

The same model that works in ort is not working in tract

called `Result::unwrap()` on an `Err` value: Evaluating #644 "main01_nonmaxsuppression11" NonMaxSuppression: output 0, expected N,3,?, got 0,3,I64

Please note that I can't make changes to model as it is a public model and I can only do inferencing.

**Comment by kali** [comment]
It may be a tract bug, but it's hard to say without the model, the code and/or the input :)

**Comment by theyash14** [comment]
@kali, Thanks for replying.
https://github.com/Kazuhito00/NARUTO-HandSignDetection/tree/main/model/yolox

There are 2 models (.onnx files), **yolox_nano_with_post** (This model outputs the mentioned error when I try to load model)  
**yolox_nano** (Loads fine, but result is different from ort)

Please let me know if you need any details. Below is my model loading and inference code.

```

let model = tract_onnx::onnx().
        // load the model
        .model_for_path(model_path)
        .unwrap()
        //.into_optimized()
        //.unwrap()
        // make the model runnable and fix its inputs and outputs
        .into_runnable()
        .unwrap();

    println!("Model loaded");

    let input: Array4<f32> =
        tract_ndarray::Array4::from_shape_vec((1, 3, 416, 416), input_image.into_raw_vec())
            .unwrap();

    let tensor_input = Tensor::from(input);

    let result = model.run(tvec!(tensor_input.into())).unwrap();
    let outputs = result[0].to_array_view::<f32>().unwrap().to_owned(); // Ensure you have the correct shape

```

Commented the optimized method as I was getting error for one model and no difference in value for the other model.
